header/header.go

```diff
@@ -94,7 +97,22 @@ func (eh *ExtendedHeader) Time() time.Time {
 // NOTE: It purposely overrides Hash method of RawHeader to get it directly from Commit without
 // recomputing.
```

Comment:
Well, practically we're recomputing it but on the other side we do this when it's not present. Any ideas for the best possible doc-comment here? (or just drop the part about recomputing, lol)

Reply:
And added a fresh one

Reply:
Done